src/datepicker/datepicker.component.ts

```diff
@@ -271,14 +271,17 @@ export class DatePicker implements OnDestroy, OnChanges, AfterViewChecked {
 	 */
 	protected updateClassNames() {
 		if (!this.elementRef) { return; }
-		document.querySelector(".flatpickr-calendar").addEventListener("click", (event) => { event.stopPropagation(); });
 		// get all the possible flatpickrs in the document - we need to add classes to (potentially) all of them
 		const calendarContainer = document.querySelectorAll(".flatpickr-calendar");
 		const monthContainer = document.querySelectorAll(".flatpickr-month");
 		const weekdaysContainer = document.querySelectorAll(".flatpickr-weekdays");
 		const weekdayContainer = document.querySelectorAll(".flatpickr-weekday");
 		const daysContainer = document.querySelectorAll(".flatpickr-days");
 		const dayContainer = document.querySelectorAll(".flatpickr-day");
+		const openCalendarContainer = document.querySelectorAll(".flatpickr-calendar.open");
+
+		Array.from(openCalendarContainer)
```

Comment:
I'm not sure we need to query for only `.open` calendars ... also there may be value in removing the event listener before adding it (since we may end up adding quite a few listeners over the lifetime of the component).